File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/socialmedia/DeleteChirp.java
package com.github.dedis.popstellar.model.network.method.message.data.socialmedia;

import androidx.annotation.NonNull;

import com.github.dedis.popstellar.model.network.method.message.data.*;
import com.github.dedis.popstellar.model.objects.security.MessageID;
import com.google.gson.annotations.SerializedName;

public class DeleteChirp extends Data {

  @SerializedName("chirp_id")
  private final MessageID chirpId;

  private final long timestamp;

  /**
   * Constructor for a data Delete Chirp
   *
   * @param chirpId the id of the chirp to delete
   * @param timestamp UNIX timestamp in UTC
   */
  public DeleteChirp(MessageID chirpId, long timestamp) {
    this.chirpId = chirpId;
    this.timestamp = timestamp;
  }

  @Override
  public String getObject() {
    return Objects.CHIRP.getObject();
  }

  @Override
  public String getAction() {
    return Action.DELETE.getAction();
  }

  public MessageID getChirpId() {
    return chirpId;
  }

  public long getTimestamp() {
    return timestamp;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    DeleteChirp that = (DeleteChirp) o;
    return java.util.Objects.equals(getChirpId(), that.getChirpId())
        && java.util.Objects.equals(getTimestamp(), that.getTimestamp());
  }

  @Override
  public int hashCode() {
    return java.util.Objects.hash(getChirpId(), getTimestamp());
  }

  @NonNull
  @Override
  public String toString() {
    return "DeleteChirp{"
        + "chirpId='"
        + chirpId.getEncoded()
        + '\''
        + ", timestamp='"
        + timestamp
        + '\''
        + '}';
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/socialmedia/DeleteChirp.kt
package com.github.dedis.popstellar.model.network.method.message.data.socialmedia

import com.github.dedis.popstellar.model.Immutable
import com.github.dedis.popstellar.model.network.method.message.data.Action
import com.github.dedis.popstellar.model.network.method.message.data.Data
import com.github.dedis.popstellar.model.network.method.message.data.Objects
import com.github.dedis.popstellar.model.objects.security.MessageID
import com.google.gson.annotations.SerializedName

@Immutable
class DeleteChirp
/**
 * Constructor for a data Delete Chirp
 *
 * @param chirpId the id of the chirp to delete
 * @param timestamp UNIX timestamp in UTC
 */
(@field:SerializedName("chirp_id") val chirpId: MessageID, val timestamp: Long) : Data {

  override val `object`: String
    get() = Objects.CHIRP.`object`

  override val action: String
    get() = Action.DELETE.action

  override fun equals(other: Any?): Boolean {
    if (this === other) {
      return true
    }
    if (other == null || javaClass != other.javaClass) {
      return false
    }
    val that = other as DeleteChirp
    return chirpId == that.chirpId && timestamp == that.timestamp
  }

  override fun hashCode(): Int {
    return java.util.Objects.hash(chirpId, timestamp)
  }

  override fun toString(): String {
    return "DeleteChirp{chirpId='${chirpId.encoded}', timestamp='$timestamp'}"
  }
}
